### DISCUSSION TITLE
How to get rid of the yellow dot of microphone permission?

### DISCUSSION THREAD

**Discussion by sochalewski** [stasel/WebRTC-iOS#139] [Q&A]
Hey,

At the very beginning I'd like to thank you for this demo, it's really a great resource!

I've been fighting an uneven battle with the problem described in the title for some time now. The application I build only occasionally needs microphone privileges (by default the client listens, only sending its audio on explicit request), and despite this the user sees a yellow dot in the status bar all the time. This can raise quite a few questions about privacy.

Today I tried making the appropriate changes to the code from this repository, but despite removing all references to the audio session in the WebRTCClient.swift file, the yellow dot appears as soon as the connection is established. Does anyone know any way to solve this problem?

![yellow dot](https://github.com/stasel/WebRTC-iOS/assets/5808283/211abbbe-9f11-4377-9ccc-3c84e0e304cd)

**Comment by tylerjames** [comment]
Hey, I'm having this same issue. My app only needs to receive a video/audio stream over WebRTC and doesn't ever need the microphone. But I can't figure out how to configure it so that it doesn't require microphone permissions and the audio still works.

@sochalewski did you ever sort this out?

**Reply by sochalewski** [reply]
@tylerjames Unfortunately not

**Reply by tylerjames** [reply]
That's too bad
